lib/player.js

```diff
@@ -2247,7 +2249,13 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     }
 
     // Re-filter the manifest after streams have been chosen.
-    this.filterManifestByCurrentVariant_();
+    // Note: Skip if an audio codec switching strategy is being enforced.
+    if (this.config_.streaming.codecSwitchingStrategy ===
+      shaka.config.CodecSwitchingStrategy.DISABLED) {
+      this.filterManifestByCurrentVariant_();
+    } else {
+      shaka.log.debug('Skipping re-filter : codec switching strategy');
```

Comment:
I'm not super happy with this debug log either. It could be more specific.
'Skipping re-filter. Not necessary due to codec switching strategy."

Reply:
Updated per request.